crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -690,9 +699,21 @@ where
             ) => {
                 debug_assert_eq!(&self.current_assignments, &[]);
                 self.visit_expr(value);
-                self.push_assignment(aug_assign.into());
-                self.visit_expr(target);
-                self.pop_assignment();
+
+                // See https://docs.python.org/3/library/ast.html#ast.AugAssign
+                if matches!(
+                    **target,
+                    ast::Expr::Attribute(_)
+                        | ast::Expr::Subscript(_)
+                        | ast::Expr::Starred(_)
+                        | ast::Expr::Name(_)
```

Comment:
I tried to search for something like `Expr::is_valid_{aug_,ann_,}assign_target` methods but couldn't find anything. And all existing code that deals with assignments could not make use of such a function, as it typically needs to do something specific for each branch. So it's just an inline match for now.